### PR TITLE
Update buf, connect-go, and golangci-lint

### PR DESCRIPTION
**Pull request by pkwarren**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ linters:
     - containedctx
     - contextcheck
     - decorder
-    - depguard
+    # - depguard
```
Comment (pkwarren) [inline .golangci.yml:20]: Depguard now fails on all non-stdlib dependencies by default. If we want to configure this, we will want to tweak it separately in each downstream project. For now, I've disabled the linter.

File: .golangci.yml [hunk 2]
```diff
@@ -32,13 +32,15 @@ linters:
     - goprintffuncname
     - gosec
     - gosimple
+    - gosmopolitan
```
Comment (pkwarren) [inline .golangci.yml:35]: New linter for i18n checks.

File: .golangci.yml [hunk 3]
```diff
@@ -32,13 +32,15 @@ linters:
...
     - govet
     - grouper
     - importas
     - ineffassign
     - interfacebloat
     - logrlint
     - makezero
+    - mirror
```
Comment (pkwarren) [inline .golangci.yml:43]: Checks for unnecessary string/[]byte conversions when methods exist that don't require a conversion.

File: .golangci.yml (hunk 2 above)
Comment (unmultimedio) [reply .golangci.yml:35]: Where do we have i18n logic?

**Review by unmultimedio — APPROVED** [review]

File: .golangci.yml (hunk 2 above)
Comment (pkwarren) [reply .golangci.yml:35]: We don't but apparently this linter helps to catch bad patterns that prevent adding it later.

**Pull request merged**